subscriptions.md

```diff
@@ -0,0 +1,309 @@
+
+The supported parameters are:
+* page, size [see pagination](README.md#Pagination)
+* eperson_id: mandatory, the uuid of the eperson that benefit of content subscription
+* resource: mandatory, the uuid of the DSpaceObject (Community, Collection, Item)
+
+```json
+{
+  "subscriptions": [
+    {
+      "id": 60,
+      "type": "subscription",
+      "subscriptionParameterList" : [ {
+			    "name" : "frequency",
+			    "value" : "D"
+		  } ],
+      "subscriptionType": "content",
+      "_links": {
+        "resource": {
+          "href": "http://localhost:8080/server/api/core/subscriptions/60/resource"
+        },
+        "eperson": {
+          "href": "http://localhost:8080/server/api/core/subscriptions/60/eperson"
+        },
+        "self": {
+          "href": "http://localhost:8080/server/api/core/subscriptions/60"
+        }
+      }
+    }
+  ],
+  "_links": {
+    "self": {
+      "href": "http://localhost:8080/server/api/core/subscriptions/search/findByEPerson?id=49e5ff01-5467-4acd-be21-9ad7374be929"
+    }
+  },
+  "page": {
+    "size": 20,
+    "totalElements": 1,
+    "totalPages": 1,
+    "number": 0
+  }
+}
+```
+Status codes:
+* 200 OK - if the operation succeed
+* 400 Bad Request - if the resource or eperson_id parameters are missing or invalid
+* 401 Unauthorized - if you are not authenticated
+* 403 Forbidden - if you are not logged in with sufficient permissions. Only system administrators or the user specified in the eperson_id parameter can use the endpoint
+
+## Creating a subscription
+**POST /api/core/subscriptions?eperson_id=<:uuid>&resource=<:uuid>**
+
+Request body:
+
+```json
+{
+  "subscriptionType": "content",
+  "subscriptionParameterList": [
+    {
+      "name": "frequency",
+      "value": "D"
+    },
+    {
+      "name": "frequency",
+      "value": "W"
+    }
+  ]
+}
+```
+The parameters are:
+eperson_id: (mandatory), the uuid of the eperson that will be grant of the subscription.
+resource: (mandatory), the uuid of the resource target of the subscription
+
+The json body must be valid that mean:
+- subscriptionType must be 'content'
+- name must be 'frequency'
+- value must be one of the following values 'D' stand for Day, 'W' stand for Week and 'M' stand for Month
+
+Return codes:
+* 200 OK - if the operation succeed, the created subscription is returned
+* 400 Bad Request - if one or both the resource and the eperson_id parameters are missing or aren't uuid
+* 401 Unauthorized - if you are not authenticated
+* 403 Forbidden - if you are not logged in with sufficient permissions. Only system administrators and Owner can create subscription
+* 422 Unprocessable Entity - if the subscriptionType or subscriptionParameter name or value are invalid
+
+## Updating subscription
+** PUT /api/core/subscription/<:id>**
+
+It is possible to update a subscription with id
+`curl -X PUT '{dspace7-url}/api/core/subscriptions/{id}?resource={id}&eperson_id={id}
```

Comment:
I noted this in my latest review of REST PR, but I think this `PUT` may have a major flaw.. It should not be possible to change the `resource` or `eperson_id` associated with the subscription, as that can be a security issue.  This endpoint should NOT accept parameters.  It should only allow for updates to the JSON *body*